### PR TITLE
binary-search : add test template

### LINKED ISSUE #1930: binary-search: add test template
## Task
Create a test template file for `binary-search`

### Details
1. If this is your first time contributing to this repository, welcome!
    - Contributing to a repository on GitHub requires a [fork](https://help.github.com/en/articles/fork-a-repo)
1. Create a branch on your local repository called `binary-search-add-test-template`
    - Run: `git checkout -b binary-search-add-test-template`
1. Clone problem-specifications repository; this is used to populate the template
    - Run: `git clone https://github.com/exercism/problem-specifications ../problem-specifications`
1. Create `exercises/binary-search/.meta/template.j2`
    - See [Generator Documentation] for more information
1. Generate test file from new template
    - Run: `bin/generate_tests.py binary-search`

At this point, if the generator fails, or you are otherwise stuck,
please feel free to reach out! It is perfectly acceptable to submit
an incomplete PR in order to get feedback or assistance.

### Resources
- [Contributing Guide]
- [Generator Documentation]
- [Closing issues using keywords]

[Contributing Guide]: https://github.com/exercism/python/blob/master/docs/CONTRIBUTING.md
[Generator Documentation]: https://github.com/exercism/python/blob/master/docs/GENERATOR.md
[Closing issues using keywords]: https://help.github.com/en/articles/closing-issues-using-keywords

### PR DESCRIPTION
**Pull request by mpatibandla**
Resolves #1930

### CODE REVIEW

File: exercises/binary-search/.meta/template.j2 [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{%- import "generator_macros.j2" as macros with context -%}
+
+{%- macro test_call(case) %}
+            {{ case["property"] }}(
+                {% for arg in case["input"].values() -%}
+                "{{ arg }}",
+                {% endfor %}
```
Comment (cmccandless) [inline exercises/binary-search/.meta/template.j2]: As we don't have any guarantee that the order of these arguments won't change in the canonical-data, might I suggest:
```
{{ case["property"] }}(
    case["input"]["array"],
    case["input"]["value"]
)
```

File: exercises/binary-search/.meta/template.j2 [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+class {{ exercise | camel_case }}Test(unittest.TestCase):
+    {% for case in cases -%}
+    def test_{{ case["description"] | to_snake }}(self):
+        {%- if case is error_case %}
+        with self.assertRaisesWithMessage(ValueError):
+            {{- test_call(case) }}
+        {%- else %}
+        expected = {{ case["expected"] }}
+        self.assertEqual({{- test_call(case) }}, expected)
```
Comment (cmccandless) [inline exercises/binary-search/.meta/template.j2]: The test file diff will be much easier to review if you can embed the `{{ case["expected"] }}` directlyin in the self.assertEqual:
```
self.assertEqual({{- test_call(case) }}, {{ case["expected"] }})
```

File: exercises/binary-search/.meta/template.j2 (hunk 1 above)
Comment (cmccandless) [reply exercises/binary-search/.meta/template.j2]: The other issue with the current changes is that neither the `array` input nor the `value` input are a string, but the for loop there treats them as such.

Comment (mpatibandla) [reply exercises/binary-search/.meta/template.j2]: Changes made as suggested. Thanks.

File: exercises/binary-search/.meta/template.j2 (hunk 2 above)
Comment (mpatibandla) [reply exercises/binary-search/.meta/template.j2]: Updated accordingly.

File: exercises/binary-search/.meta/template.j2 [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        with self.assertRaisesWithMessage(ValueError):
+            {{- test_call(case) }}
+        {%- else %}
+        self.assertEqual({{- test_call(case) }}, {{ case["expected"] }})
+        {%- endif %}
+    {% endfor %}
+
+    {{ macros.utility() }}
+
+{{ macros.footer() }}
```
Comment (cmccandless) [inline exercises/binary-search/.meta/template.j2]:
```suggestion
{{ macros.footer(has_error_case) }}
```

File: exercises/binary-search/.meta/template.j2 [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    def test_{{ case["description"] | to_snake }}(self):
+        {%- if case is error_case %}
+        with self.assertRaisesWithMessage(ValueError):
+            {{- test_call(case) }}
+        {%- else %}
+        self.assertEqual({{- test_call(case) }}, {{ case["expected"] }})
+        {%- endif %}
+    {% endfor %}
+
+    {{ macros.utility() }}
```
Comment (cmccandless) [inline exercises/binary-search/.meta/template.j2]:
```suggestion
```

**Comment by mpatibandla** [comment]
updated!

**Pull request merged**